egret/models/unit_commitment.py

```diff
@@ -599,6 +599,25 @@ def solve_unit_commitment(model_data,
         commitment_dict = {}
         commitment_cost_dict = {}
         production_cost_dict = {}
+        ramp_up_avail_dict = {}
+
+        ## all of the potential constraints that could limit maximum output
+        ## Not all unit commitment models have these constraints, so first
+        ## we need check if they're on the model object
+        ramp_up_avail_potential_constrs = [
+                                          'EnforceMaxAvailableRampUpRates',
+                                          'AncillaryServiceRampUpLimit',
+                                          'power_limit_from_start',
+                                          'power_limit_from_stop',
+                                          'power_limit_from_start_stop',
+                                          'power_limit_from_start_stops',
+                                          'EnforceMaxAvailableRampDownRates',
+                                          'EnforceMaxCapacity',
+                                         ]
+        ramp_up_avail_constrs = []
+        for constr in ramp_up_avail_potential_constrs:
+            if hasattr(m, constr):
```

Comment:
Is using ```hasattr``` the best way to do this? What is used to decide if these constraints are added to the model in the first place? Is it the ```options``` ```dict``` that the user passes into this function? If so, why not continue to use it?

Reply:
On the same note, have you ever used ```ConfigBlock```/```ConfigValue``` from Pyomo? Given the increasing complexity of this unit commitment model, they may be useful.

Reply:
`hasattr` may not be the best way to do this, but it is the most straightforward. You're correct that the same information is available based on the model the user specifies. But, I think inferring which constraints are used based on that would involve a rather complicated if statement, for which it would be harder to be sure if I covered every case.

I'll have a look at `ConfigBlock`/`ConfigValue`, as obviously there is room for improvement here.

Reply:
Understood.

Reply:
It would probably be worth somehow encapsulating the logic for determining the potential universe of ramp-up available constraints. Seems like a check / set that might be more generally useful.